src/test/java/com/google/cloud/tools/libraries/LibrariesTest.java

```diff
@@ -36,27 +65,59 @@ public void testWellFormed() throws ParserConfigurationException, SAXException,
-    JsonReader reader = factory.createReader(in); 
-    Iterator<JsonValue> apis = reader.readArray().iterator();
-    Assert.assertTrue(apis.hasNext());
-    for (JsonObject api = (JsonObject) apis.next(); apis.hasNext(); api = (JsonObject) apis.next()) {
-    
-      new URI(api.getString("documentation"));
-      if (api.getString("icon") != null) {
-        new URI(api.getString("icon"));
+  public void testJson() throws URISyntaxException {
+    Assert.assertTrue(apis.length > 0);
+    for (int i = 0; i < apis.length; i++) { 
+      assertApi(apis[i]);
+    }
+  }
+
+  private static final String[] statuses = {"alpha", "beta", "GA"};
+
+  private static void assertApi(JsonObject api) throws URISyntaxException {
+    Assert.assertFalse(api.getString("name").isEmpty());
+    Assert.assertFalse(api.getString("description").isEmpty());
+    String transport = api.getString("transport");
+    Assert.assertTrue(transport + " is not a recognized transport",
+        "http".equals(transport) || "grpc".equals(transport));
+    new URI(api.getString("documentation"));
+    if (api.getString("icon") != null) {
+      new URI(api.getString("icon"));
+    }
+    JsonArray clients = api.getJsonArray("clients");
+    Assert.assertFalse(clients.isEmpty());
+    for (int i = 0; i < clients.size(); i++) {
+      JsonObject client = (JsonObject) clients.get(i);
+      String status = client.getString("status");
+      Assert.assertThat(statuses, hasItemInArray(status));
+      new URI(client.getString("apireference"));
+      new URI(client.getString("site"));
+      Assert.assertTrue(client.getString("languageLevel").matches("1\\.\\d+\\.\\d+"));
+      Assert.assertFalse(client.getString("name").isEmpty());
+      Assert.assertNotNull(client.getJsonObject("mavenCoordinates"));
+      if (client.getString("source") != null) {
+        new URI(client.getString("source"));
+      }
+    }
+  }
+  
+  @Test
+  public void testDuplicates() throws URISyntaxException {
+    Map<String, String> apiCoordinates = new HashMap<>();
+    for (int i = 0; i < apis.length; i++) { 
+      JsonObject api = apis[i];
+      String name = api.getString("name");
+      if (apiCoordinates.containsKey(name)) {
+        Assert.fail(name + " is defined twice");
       }
-      JsonArray clients = api.getJsonArray("clients");
-      Assert.assertFalse(clients.isEmpty());
-      for (int i = 0; i < clients.size(); i++) {
-        JsonObject client = (JsonObject) clients.get(i);
-        String status = client.getString("status");
-        Assert.assertTrue(status, 
-            "beta".equals(status) || "alpha".equals(status) || "GA".equals(status));
+      JsonObject coordinates =
+          ((JsonObject) api.getJsonArray("clients").get(0)).getJsonObject("mavenCoordinates");
+      String mavenCoordinates = 
+          coordinates.getString("groupId") + ":" + coordinates.getString("artifactId");
+      if (apiCoordinates.containsValue(mavenCoordinates)) {
```

Comment:
It should be ok to provide different versions of an API library, shouldn't it?  E.g., Java libraries for BigQuery 1.0.0 vs 2.0.0?

Reply:
I think that would have a different group or artifact ID. I don't think we want both 1.3.0 and 1.4.0.